python/src/com/jetbrains/python/documentation/PyTypeModelBuilder.java

```diff
@@ -197,7 +197,16 @@ public TypeModel build(@Nullable PyType type,
     myVisited.put(type, null); //mark as evaluating
 
     TypeModel result = null;
-    if (type instanceof PyCollectionType) {
+    if (type instanceof PyTupleType) {
+      final List<TypeModel> elementModels = new ArrayList<TypeModel>();
+      final PyTupleType tupleType = (PyTupleType)type;
+      for (int i = 0; i < (tupleType.isHomogeneous() ? 1 : tupleType.getElementCount()); i++) {
+        final PyType elementType = tupleType.getElementType(i);
+        elementModels.add(build(elementType, true));
+      }
+      result = new TupleType(elementModels, tupleType.isHomogeneous());
+    }
+    else if (type instanceof PyCollectionType) {
```

Comment:
It's better to create test or tests for changes which could change PyCharm behaviour